### PR TITLE
CppCheck HeatBalanceSurfaceManager

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes CppCheck issues
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 1]
```diff
@@ -3027,7 +3001,7 @@ void InitSolarHeatGains(EnergyPlusData &state)
                  state.dataEnvrn->DifSolarRad * SurfIncSolarMultiplier * state.dataSolarShading->SurfAnisoSkyMult(OutShelfSurf)) *
                 state.dataDaylightingDevicesData->Shelf(ShelfNum).OutReflectSol;
 
-            Real64 GndReflSolarRad = state.dataEnvrn->GndSolarRad * SurfIncSolarMultiplier;
+            GndReflSolarRad = state.dataEnvrn->GndSolarRad * SurfIncSolarMultiplier;
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:2978]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:3030]:(style),[shadowVariable],Local variable 'GndReflSolarRad' shadows outer variable

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 2]
```diff
@@ -2460,18 +2445,6 @@ void InitSolarHeatGains(EnergyPlusData &state)
...
-    using DaylightingDevices::TransTDD;
-    using General::POLYF;
-    using SolarShading::CalcInteriorSolarDistribution;
-    using namespace DataWindowEquivalentLayer;
-    using SolarShading::SurfaceScheduledSolarInc;
-    using SolarShading::WindowScheduledSolarAbs;
-
-    auto &AbsDiffWin = state.dataHeatBalSurfMgr->AbsDiffWin;
-    auto &AbsDiffWinGnd = state.dataHeatBalSurfMgr->AbsDiffWinGnd;
-    auto &AbsDiffWinSky = state.dataHeatBalSurfMgr->AbsDiffWinSky;
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:2472]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:2471]:(style),[variableScope],The scope of the variable 'AbsDiffWin' can be reduced.
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:2472]:(style),[variableScope],The scope of the variable 'AbsDiffWinGnd' can be reduced.
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:2473]:(style),[variableScope],The scope of the variable 'AbsDiffWinSky' can be reduced.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 3]
```diff
@@ -3623,7 +3610,7 @@ void InitSolarHeatGains(EnergyPlusData &state)
         if (state.dataSurface->UseRepresentativeSurfaceCalculations) {
             for (int zoneNum = 1; zoneNum <= state.dataGlobal->NumOfZones; ++zoneNum) {
                 for (int spaceNum : state.dataHeatBal->Zone(zoneNum).spaceIndexes) {
-                    auto &thisSpace = state.dataHeatBal->space(spaceNum);
+                    auto const &thisSpace = state.dataHeatBal->space(spaceNum);
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:3587]: Lots of these:

[src/EnergyPlus/HeatBalanceSurfaceManager.cc:3626]:(style),[constVariable],Variable 'thisSpace' can be declared as reference to const

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 4]
```diff
@@ -4446,17 +4430,13 @@ void ComputeDifSolExcZonesWIZWindows(EnergyPlusData &state, int const NumberOfEn
...
-    //       DATE WRITTEN
     //       MODIFIED       Jun 2007 - Lawrie - Speed enhancements.
     //       RE-ENGINEERED  Winkelmann, Lawrie
 
     // PURPOSE OF THIS SUBROUTINE:
     // This subroutine computes the diffuse solar exchange factors between zones with
     // interzone windows.
 
-    auto &Surface = state.dataSurface->Surface;
-
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:4458]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:4458]:(style),[constVariable],Variable 'Surface' can be declared as reference to const

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 5]
```diff
@@ -5680,14 +5618,6 @@ void CalcThermalResilience(EnergyPlusData &state)
 void ReportThermalResilience(EnergyPlusData &state)
 {
 
-    int HINoBins = 5;                     // Heat Index range - number of bins
-    int HumidexNoBins = 5;                // Humidex range - number of bins
-    int SETNoBins = 5;                    // SET report column numbers
-    int ColdHourOfSafetyNoBins = 5;       // Cold Stress Hour of Safety number of columns
-    int HeatHourOfSafetyNoBins = 5;       // Heat Stress Hour of Safety number of columns
-    int UnmetDegreeHourNoBins = 6;        // Unmet Degree Hour number of columns
-    int DiscomfortWtExceedHourNoBins = 4; // Unmet Degree Hour number of columns
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:5688]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:5683]:(style),[variableScope],The scope of the variable 'HINoBins' can be reduced.
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5684]:(style),[variableScope],The scope of the variable 'HumidexNoBins' can be reduced.
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5685]:(style),[variableScope],The scope of the variable 'SETNoBins' can be reduced.
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5686]:(style),[variableScope],The scope of the variable 'ColdHourOfSafetyNoBins' can be reduced.
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5687]:(style),[variableScope],The scope of the variable 'HeatHourOfSafetyNoBins' can be reduced.
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5688]:(style),[variableScope],The scope of the variable 'UnmetDegreeHourNoBins' can be reduced.
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5689]:(style),[variableScope],The scope of the variable 'DiscomfortWtExceedHourNoBins' can be reduced.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 6]
```diff
@@ -5932,11 +5871,6 @@ void ReportThermalResilience(EnergyPlusData &state)
                         }
                     }
 
-                    Real64 VeryHotPMVThresh = 3.0;
-                    Real64 WarmPMVThresh = 0.7;
-                    Real64 CoolPMVThresh = -0.7;
-                    Real64 VeryColdPMVThresh = -3.0;
-                    Real64 PMV = state.dataThermalComforts->ThermalComfortData(iPeople).FangerPMV;
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:5938]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:5935]:(style),[shadowVariable],Local variable 'VeryHotPMVThresh' shadows outer variable
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5936]:(style),[shadowVariable],Local variable 'WarmPMVThresh' shadows outer variable
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5937]:(style),[shadowVariable],Local variable 'CoolPMVThresh' shadows outer variable
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5938]:(style),[shadowVariable],Local variable 'VeryColdPMVThresh' shadows outer variable
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:5939]:(style),[shadowVariable],Local variable 'PMV' shadows outer variable

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 7]
```diff
@@ -6233,10 +6167,6 @@ void ReportThermalResilience(EnergyPlusData &state)
                         }
                     }
 
-                    Real64 Temperature = state.dataZoneTempPredictorCorrector->zoneHeatBalance(ZoneNum).ZTAV;
-                    Real64 CoolingSetpoint = state.dataHeatBalFanSys->ZoneThermostatSetPointHi(ZoneNum);
-                    Real64 HeatingSetpoint = state.dataHeatBalFanSys->ZoneThermostatSetPointLo(ZoneNum);
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:6237]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:6236]:(style),[shadowVariable],Local variable 'Temperature' shadows outer variable
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:6237]:(style),[shadowVariable],Local variable 'CoolingSetpoint' shadows outer variable
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:6238]:(style),[shadowVariable],Local variable 'HeatingSetpoint' shadows outer variable

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 8]
```diff
@@ -6261,8 +6191,8 @@ void ReportThermalResilience(EnergyPlusData &state)
 
 void ReportCO2Resilience(EnergyPlusData &state)
 {
-    int NoBins = 3;
     if (state.dataHeatBalSurfMgr->reportCO2ResilienceFirstTime) {
+        int NoBins = 3;
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:6161]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:6264]:(style),[variableScope],The scope of the variable 'NoBins' can be reduced.
[src/EnergyPlus/HeatBalanceSurfaceManager.cc:6355]:(style),[variableScope],The scope of the variable 'NoBins' can be reduced.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 9]
```diff
@@ -7527,9 +7456,9 @@ void CalcHeatBalanceOutsideSurf(EnergyPlusData &state,
 
 Real64 GetQdotConvOutPerArea(EnergyPlusData &state, int const SurfNum)
 {
-    auto &Surface = state.dataSurface->Surface;
-    int OPtr = Surface(SurfNum).OSCMPtr;
-    if (Surface(SurfNum).OSCMPtr > 0) { // Optr is set above in this case, use OSCM boundary data
+    auto const &surface = state.dataSurface->Surface(SurfNum);
+    int OPtr = surface.OSCMPtr;
+    if (surface.OSCMPtr > 0) { // Optr is set above in this case, use OSCM boundary data
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:7444]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:7530]:(style),[constVariable],Variable 'Surface' can be declared as reference to const

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 10]
```diff
@@ -8179,11 +8115,11 @@ void CalcHeatBalanceInsideSurf2(EnergyPlusData &state,
...
                             if (ConstrNumSh != 0) {
-                                auto const *thisMaterial = dynamic_cast<Material::MaterialChild *>(
+                                auto const *thisMaterial2 = dynamic_cast<Material::MaterialChild *>(
                                     state.dataMaterial->Material(state.dataConstruction->Construct(ConstrNumSh).LayerPoint(1)));
-                                assert(thisMaterial != nullptr);
-                                RoughSurf = thisMaterial->Roughness;
-                                EmisOut = thisMaterial->AbsorpThermal;
+                                assert(thisMaterial2 != nullptr);
+                                RoughSurf = thisMaterial2->Roughness;
+                                EmisOut = thisMaterial2->AbsorpThermal;
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:8103]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:8182]:(style),[shadowVariable],Local variable 'thisMaterial' shadows outer variable

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 11]
```diff
@@ -8866,22 +8804,22 @@ void CalcHeatBalanceInsideSurf2CTFOnly(EnergyPlusData &state,
...
                                     if (ConstrNumSh != 0) {
-                                        auto const *thisMaterial = dynamic_cast<Material::MaterialChild *>(
+                                        auto const *thisMaterial2 = dynamic_cast<Material::MaterialChild *>(
                                             state.dataMaterial->Material(state.dataConstruction->Construct(ConstrNumSh).LayerPoint(1)));
-                                        assert(thisMaterial != nullptr);
-                                        RoughSurf = thisMaterial->Roughness;
-                                        EmisOut = thisMaterial->AbsorpThermal;
+                                        assert(thisMaterial2 != nullptr);
+                                        RoughSurf = thisMaterial2->Roughness;
+                                        EmisOut = thisMaterial2->AbsorpThermal;
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:8802]: [src/EnergyPlus/HeatBalanceSurfaceManager.cc:8880]:(style),[shadowVariable],Local variable 'thisMaterial' shadows outer variable

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 12]
```diff
@@ -2305,26 +2287,27 @@ void InitThermalAndFluxHistories(EnergyPlusData &state)
...
-                state.dataConstruction->Construct(Surface(SurfNum).Construction).UValue *
+                state.dataConstruction->Construct(surface.Construction).UValue *
                 (state.dataHeatBalSurf->SurfOutsideTempHist(1)(SurfNum) - state.dataHeatBalSurf->SurfInsideTempHist(1)(SurfNum));
             state.dataHeatBalSurf->SurfInsideFluxHist(CTFTermNum)(SurfNum) = state.dataHeatBalSurf->SurfOutsideFluxHist(2)(SurfNum);
         }
     }
-    for (SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
-
-        if (!Surface(SurfNum).HeatTransSurf) continue; // Skip non-heat transfer surfaces
+    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
```
Comment (amirroth) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc]: There is a list of heat transfer surfaces. Do we want to create a list of ExtCavity surfaces also?

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 13]
```diff
@@ -2338,34 +2321,36 @@ void InitThermalAndFluxHistories(EnergyPlusData &state)
         state.dataSurfaceGeometry->kivaManager.surfaceConvMap[SurfNum].out = KIVA_CONST_CONV(0.0);
     }
     if (!state.dataHeatBal->SimpleCTFOnly || state.dataGlobal->AnyEnergyManagementSystemInModel) {
-        for (SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
-            if (!Surface(SurfNum).HeatTransSurf) continue;
-            if ((Surface(SurfNum).ExtBoundCond == ExternalEnvironment) || (Surface(SurfNum).ExtBoundCond == OtherSideCondModeledExt)) {
+        for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+            auto &surface = state.dataSurface->Surface(SurfNum);
+            if (!surface.HeatTransSurf) continue;
```
Comment (amirroth) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc]: Use list of heat transfer surfaces.

Comment (rraustad) [reply src/EnergyPlus/HeatBalanceSurfaceManager.cc]: `if (!surface.HeatTransSurf) continue` shows up in 4 other places in this file.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 14]
```diff
@@ -2298,25 +2294,15 @@ void InitThermalAndFluxHistories(EnergyPlusData &state)
...
     }
-    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
-        auto const &surface = state.dataSurface->Surface(SurfNum);
-        if (!surface.HeatTransSurf) continue; // Skip non-heat transfer surfaces
-
-        if (state.dataSurface->SurfExtCavityPresent(SurfNum)) {
-            state.dataHeatBal->ExtVentedCavity(state.dataSurface->SurfExtCavNum(SurfNum)).TbaffleLast = 20.0;
-            state.dataHeatBal->ExtVentedCavity(state.dataSurface->SurfExtCavNum(SurfNum)).TairLast = 20.0;
-        }
-    }
```
Comment (rraustad) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc]: moved to 2323

File: src/EnergyPlus/UnitarySystem.cc [hunk 15]
```diff
@@ -4545,7 +4545,7 @@ namespace UnitarySystems {
                         this->m_CondenserNodeNum = thisCoolCoil.CondenserInletNodeNum(1);
 
                         if (this->m_FanExists) {
-                            thisCoolCoil.SupplyFanName = loc_m_FanName;
+                            thisCoolCoil.SupplyFanName = this->m_FanName;
```
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.cc:4548]: I ran the unit tests using a debug build and this line showed a memory access issue.

**Comment by Myoldmopar** [comment]
Conflicts were a bit annoying to deal with here, but everything seems happy.  I applied format, ran unit tests and regression tests, everything is passing.  I'm considering letting CI do a full pass here, and just move onto another CppCheck branch.  I'll think about it.

**Comment by Myoldmopar** [comment]
Alright, I'm gonna go ahead and merge here since the other CppCheck branch has some comments to address.  Thanks @rraustad !

**Pull request merged**